Remove default font styles from cookie banner Sass (#2432)
alphagov/govuk-frontend: CHANGELOG.md

This change was introduced in [pull request #2417: Remove deprecated `govuk-tag--inactive class`](https://github.com/alphagov/govuk-frontend/pull/2417).

> #### Style the `html` in your cookie banner

Reviewer: Are we referring here to HTML-the-thing or to the `html` parameter? If it's the thing, then we could just write HTML here.

```suggestion
#### Style the HTML in your cookie banner
```

Author: Hm, kind of both 😅 It's if someone uses the `html` parameter to pass HTML code in, rather than using the `text`parameter and just passing in a string of text.

Author: Although actually, thinking about it, I think the same applies if a user is using the HTML version of the component rather than the macros. So maybe we need to be more generic and just talk about any custom HTML a user provides... 

I'll have a go at tweaking it and see what I come up with.

Author: @EoinShaughnessy I've had a go at rewording this a bit to be more about "custom HTML" and less about how the user is providing that custom HTML. Would you mind taking another look?

> We have removed the default font styles from the cookie banner Sass, to make it more obvious when classes/styles have been forgotten.

Reviewer: ```suggestion
We've removed the default font styles from the cookie banner Sass, to make it more obvious when classes and styles have been forgotten.
```
Forgotten by what? (Sorry if daft question - let me know if the meaning would be obvious to users!)

Author: This is talking about classes/styling being forgotten on the HTML that's being passed into the cookie banner. Some users may assume that the HTML they pass in is somehow automatically styled. This isn't the case so we want to make it really obvious if users have forgotten to add their own styles.

> If you are passing `html` into the cookie banner component, you must make sure you are applying the appropriate classes/styles to that HTML, such as adding the `govuk-body` class to any paragraph tags. You must do this to prevent your cookie banner displaying with unstyled text.

Reviewer: ```suggestion
If you're passing `html` into the cookie banner component, make sure you're applying the appropriate classes and styles to that HTML. For example, add the `govuk-body` class to any paragraph tags. You must do this to prevent your cookie banner displaying with unstyled text.
```
Wonder could we delete "to that HTML" from the second sentence? Or is there a risk some users might think it's the cookie banner component they have to apply classes and styles to?

This change was introduced in [pull request #2417: Remove deprecated `govuk-tag--inactive class`](https://github.com/alphagov/govuk-frontend/pull/2417).

#### Style any custom HTML in your cookie banner

> We have removed the default font styles from the cookie banner Sass, to make it more obvious when classes/styles haven't been added to any custom HTML.

Reviewer: ```suggestion
We've removed the default font styles from the cookie banner Sass. This change makes it more obvious when classes and styles have not been added to any custom HTML.
```
Style guide encourages us to use positive contractions, but to avoid negative contractions (some users find them hard to read). Also says to replace '/' with 'or'.

Author: Thanks @EoinShaughnessy , really useful to know! I'll try to remember for future 😬

> If you are passing custom HTML into the cookie banner component (for example, using the `html` Nunjucks param), you must make sure you are applying the appropriate classes/styles to that HTML, such as adding the `govuk-body` class to any paragraph tags. You must do this to prevent your cookie banner displaying with unstyled text.

Reviewer: ```suggestion
##### If you're passing custom HTML (for example, using the `html` Nunjucks parameter) into the cookie banner component

Make sure you're applying the appropriate classes and styles to that HTML. For example, add the `govuk-body` class to any paragraph tags. You must do this to prevent your cookie banner displaying with unstyled text.
```
Just suggesting this to break up the sentences a little. Also took out the first "You must", in case it suffices to link the remaining "You must" with what happens if you users do not do this.

Reviewer: Is there any other way users could pass custom HTML without using the `html` param?

Author: @EoinShaughnessy If the user was using the HTML version of the component (not the Nunjucks macros) they can edit the HTML however they want and put whatever they want in there. Hopefully these users would already be more aware of styling their HTML, but it's possible someone could have added some custom HTML which previously would have looked "correct" but following this update won't look styled any more. They'd need to take the same actions as anyone using the Nunjucks macro though.

Author: I'm not 100% sure I see the benefit of splitting this out into its own heading? At least to me, it kind of suggests that there might be another heading later on with different actions if you do something else, which isn't the case?

Reviewer: That's a good point - yeah, best not to split out then.

> If you're passing custom HTML into the cookie banner component (for example, using the `html` Nunjucks parameter), you must make sure you are applying the appropriate classes/styles to that HTML, such as adding the `govuk-body` class to any paragraph tags. You must do this to prevent your cookie banner displaying with unstyled text.

Reviewer: ```suggestion
If you're passing custom HTML into the cookie banner component (for example, using the `html` Nunjucks parameter), you must make sure you are applying the appropriate classes and styles to that HTML, such as adding the `govuk-body` class to any paragraph tags. You must do this to prevent your cookie banner displaying with unstyled text.
```

> If you're passing custom HTML into the cookie banner component (for example, using the `html` Nunjucks parameter), you must make sure you are applying the appropriate classes and styles to that HTML, such as adding the `govuk-body` class to any paragraph tags. You must do this to prevent your cookie banner displaying with unstyled text.

Reviewer: ```suggestion
If you're passing custom HTML into the cookie banner component (for example, using the `html` Nunjucks parameter), you must make sure you're applying the appropriate classes and styles to that HTML, such as adding the `govuk-body` class to any paragraph tags. You must do this to prevent your cookie banner displaying with unstyled text.
```
